frontend/src/pages/RegisterInvestor.tsx

```diff
@@ -0,0 +1,245 @@
+    lastName: "",
+    firstName: "",
+    representing: [] as string[],
+    entityType: [] as string[],
+    minInvestment: "",
+    maxInvestment: "",
+    role: "investor",
+  });
+
+  const [errors, setErrors] = useState<Record<string, string>>({});
+  const [successMessage, setSuccessMessage] = useState("");
+  const [resendEmail, setResendEmail] = useState("");
+
+  const validate = () => {
+    const newErrors: Record<string, string> = {};
+
+    if (!formData.companyName) newErrors.companyName = "Не ввели назву компанії";
+    if (!formData.email) newErrors.email = "Не ввели електронну пошту";
+
+    const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*?&]).{8,}$/;
+    if (!formData.password) newErrors.password = "Не ввели пароль";
+    else if (!passwordRegex.test(formData.password)) newErrors.password = "Пароль не відповідає вимогам";
+
+    if (!formData.confirmPassword) newErrors.confirmPassword = "Не ввели пароль ще раз";
+    else if (formData.password !== formData.confirmPassword)
+      newErrors.confirmPassword = "Паролі не співпадають. Будь ласка, введіть однакові паролі в обидва поля";
+
+    if (!formData.lastName) newErrors.lastName = "Не ввели прізвище";
+    if (!formData.firstName) newErrors.firstName = "Не ввели ім’я";
+    if (!formData.representing.length) newErrors.representing = "Виберіть кого ви представляєте";
+    if (!formData.entityType.length) newErrors.entityType = "Виберіть тип суб’єкта";
+    if (!formData.minInvestment) newErrors.minInvestment = "Не ввели мінімальну інвестицію";
+    if (!formData.maxInvestment) newErrors.maxInvestment = "Не ввели максимальну інвестицію";
+
+    return newErrors;
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleCheckboxChange = (field: string, value: string) => {
+    setFormData((prev) => {
+      const updated = prev[field as keyof typeof prev] as string[];
+      if (updated.includes(value)) {
+        return { ...prev, [field]: updated.filter((v) => v !== value) };
+      } else {
+        return { ...prev, [field]: [...updated, value] };
+      }
+    });
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setErrors({});
+    setSuccessMessage("");
+
+    const validationErrors = validate();
+    if (Object.keys(validationErrors).length > 0) {
+      setErrors(validationErrors);
+      return;
+    }
+
+    try {
+      const response = await fetch("/api/auth/register/", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          email: formData.email,
+          password: formData.password,
+          role: formData.role,
+          company_name: formData.companyName,
+          investment_range_min: parseFloat(formData.minInvestment),
+          investment_range_max: parseFloat(formData.maxInvestment),
```

Comment:
Using parseFloat(formData.minInvestment) / parseFloat(formData.maxInvestment) can produce NaN if values are invalid; this may send bad payloads. Coerce with Number(...) and provide a safe fallback or validate before sending.
```
const min = Number(formData.minInvestment);
const max = Number(formData.maxInvestment);
body: JSON.stringify({ investment_range_min: isNaN(min) ? 0 : min, investment_range_max: isNaN(max) ? 0 : max }),
```